### PR TITLE
Update dnnl Add, Mul, Sub, Div ops to handle scalar values

### PR DESCRIPTION
**Pull request by georgen117**
Signed-off-by: George Nash <george.nash@intel.com>

**Description**: Describe your changes.
This updates the dnnl execution provider to handle scaler inputs and outputs properly.

**Motivation and Context**
- Why is this change required? What problem does it solve?
The dnnl execution provider was not running any Add, Mul, Sub, or Div operations that resulted in a scalar output. This resulted in over 800 nodes in MobileBert model. As well as many other models.

- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by georgen117** [comment]
@jywu-msft maintainer, please remember to run /azp run Linux DNNL CI Pipeline since this is the main pipeline that runs the dnnl ep.

**Comment by georgen117** [comment]
I don't fully understand the failures but I am almost sure it is related to this change.  The errors are showing up in the softmax cross-entropy tests.  I know how to reproduce the failure since those are failures in the onnx tests.  I will investigate.

**Comment by georgen117** [comment]
@jywu-msft can you run `/azp run Linux DNNL CI Pipeline` again

thanks.

**Comment by georgen117** [comment]
@jywu-msft could you run the other pipelines? Thanks.

**Review by jywu-msft — APPROVED** [review]
